src/parsing.jl

```diff
@@ -83,6 +101,12 @@ function parse(::Type{Interval{T}}, s::AbstractString) where T
 
     expr1 = Meta.parse(lo)
     expr2 = Meta.parse(hi)
+    if lo == "-Inf"
+        expr1 = Base.parse(Float64, lo)
+    end
+    if hi == "Inf"
+        expr2 = Base.parse(Float64, hi)
+    end
 
     interval = eval(make_interval(T, expr1, [expr2]))
```

Comment:
I don't know / remember why there's an `eval` here; there shouldn't need to be.

We should refactor the functions to avoid `eval`.